### PR TITLE
organize project files

### PR DESCRIPTION
**Pull request by lekoenig**
Restructure project files #3

### CODE REVIEW

File: 2_process/src/process_data.R [hunk 1]
```diff
@@ -0,0 +1,43 @@
+
+process_data <- function(){
+  
+  save_path <- "./2_process/out"
+  
+  # Prepare the data for plotting:
+  eval_data <- readr::read_csv(file.path("./1_fetch/out", 'model_RMSEs.csv'), col_types = 'iccd') %>%
```
Comment (jds485) [inline 2_process/src/process_data.R:7]: You can make the save path and input file path arguments to your function

File: 2_process/src/process_data.R [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                      dl_980mean = filter(eval_data, model_type == 'dl', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_980mean = filter(eval_data, model_type == 'pb', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                      dl_500mean = filter(eval_data, model_type == 'dl', exper_id == "similar_500") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_500mean = filter(eval_data, model_type == 'pb', exper_id == "similar_500") %>% pull(rmse) %>% mean %>% round(2),
+                      dl_100mean = filter(eval_data, model_type == 'dl', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_100mean = filter(eval_data, model_type == 'pb', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
+                      pgdl_2mean = filter(eval_data, model_type == 'pgdl', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_2mean = filter(eval_data, model_type == 'pb', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2))
```
Comment (jds485) [inline 2_process/src/process_data.R:31]: You should make a function for every repeated expression. So here we see the same filter command applied to each of the datasets. The filter command should be a function. That way if we ever decide to change the filter, we just change the function once

File: make_script.R [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+library(stringr)
+library(sbtools)
+library(whisker)
+
+source("./1_fetch/src/fetch_data.R")
+source("./2_process/src/process_data.R")
+source("./3_visualize/src/plot_rmse.R")
+
+# Get the data from ScienceBase
+fetch_data(save_path <- "./1_fetch/out",output_file_name <- "model_RMSEs.csv")
```
Comment (jds485) [inline make_script.R:12]: I'm curious: does this work with `<-` for argument assignment? I thought functions needed to use `=` for arguments

File: 2_process/src/process_data.R (hunk 1 above)
Comment (lekoenig) [reply 2_process/src/process_data.R]: Good suggestion - I've implemented these changes to the process_data function.

File: 2_process/src/process_data.R (hunk 2 above)
Comment (lekoenig) [reply 2_process/src/process_data.R]: OK, good idea! The "2_process" phase code now contains a filter_data function that does the same command. For this project it seems like we know the specific model types and experiment id's we're after, but I have some ideas for how we could summarize model performance without hard-coding in the model names. We could revisit down the line if useful.

File: 3_visualize/src/plot_rmse.R [hunk 4]
```diff
@@ -0,0 +1,51 @@
+
+plot_rmse <- function(data){
+  
+
+  save_path <- "./3_visualize/out"
```
[the remark this answers is not in the document]
Comment (lekoenig) [reply 3_visualize/src/plot_rmse.R]: The plot_rmse function now contains arguments to designate the save_path.

File: make_script.R [hunk 5]
```diff
@@ -0,0 +1,29 @@
+library(dplyr)
+library(readr)
+library(stringr)
+library(sbtools)
+library(whisker)
+
+source("./1_fetch/src/fetch_data.R")
```
[the remark this answers is not in the document]
Comment (lekoenig) [reply make_script.R:7]: Thanks, I didn't realize that!

**Pull request merged**